File: src/main/java/nl/crosscode/x509certbuddy/ui/CertEditorElementRender.java
package nl.crosscode.x509certbuddy.ui;

import com.intellij.openapi.editor.EditorCustomElementRenderer;
import com.intellij.openapi.editor.Inlay;
import com.intellij.openapi.editor.markup.TextAttributes;
import org.jetbrains.annotations.NotNull;

import java.awt.*;
import java.awt.geom.Rectangle2D;
import java.security.cert.X509Certificate;

public class CertEditorElementRender implements EditorCustomElementRenderer {
    private final X509Certificate certificate;
    private Font font;


    public CertEditorElementRender(X509Certificate certificate) {
        this.certificate = certificate;
    }

    @Override
    public int calcWidthInPixels(@NotNull Inlay inlay) {
        return 600;
    }
/*
    @Override
    public int calcHeightInPixels(@NotNull Inlay inlay) {
        return 20;
    }
*/
    @Override
    public void paint(@NotNull Inlay inlay, @NotNull Graphics2D g, @NotNull Rectangle2D targetRegion, @NotNull TextAttributes textAttributes) {
     //   g.get
        if (font==null) {
            font = g.getFont().deriveFont(g.getFont().getStyle(), g.getFont().getSize()-1);
        }
        Font oldFont = g.getFont();
        g.setFont(font);
        int x = (int)inlay.getEditor().offsetToPoint2D(inlay.getOffset()).getX();
        g.setColor(Color.gray);
         g.drawString(certificate.getSubjectDN().getName() + " (0x"+certificate.getSerialNumber().toString(16)+")", (int) targetRegion.getX()+1+x, (int) targetRegion.getMaxY()-2);
        EditorCustomElementRenderer.super.paint(inlay, g, targetRegion, textAttributes);
        g.setFont(oldFont);
    }

}

File: src/main/kotlin/nl/crosscode/x509certbuddy/ui/CertEditorElementRender.kt
package nl.crosscode.x509certbuddy.ui

import com.intellij.openapi.editor.EditorCustomElementRenderer
import com.intellij.openapi.editor.Inlay
import com.intellij.openapi.editor.markup.TextAttributes
import com.intellij.ui.JBColor
import java.awt.Font
import java.awt.Graphics2D
import java.awt.geom.Rectangle2D
import java.security.cert.X509Certificate

class CertEditorElementRender(private val certificate: X509Certificate) : EditorCustomElementRenderer {
    private var font: Font? = null


    override fun calcWidthInPixels(inlay: Inlay<*>): Int {
        return 600
    }

    override fun paint(inlay: Inlay<*>, g: Graphics2D, targetRegion: Rectangle2D, textAttributes: TextAttributes) {
        if (font == null) {
            font = g.font.deriveFont(g.font.style, (g.font.size - 1).toFloat())
        }
        val oldFont = g.font
        g.font = font
        val x = inlay.editor.offsetToPoint2D(inlay.offset).x.toInt()
        g.color = JBColor.GRAY
        g.drawString(
            certificate.subjectX500Principal.name + " (0x" + certificate.serialNumber.toString(16) + ")",
            targetRegion.x.toInt() + 1 + x,
            targetRegion.maxY.toInt() - 2
        )
        super.paint(inlay, g, targetRegion, textAttributes)
        g.font = oldFont
    }
}
